src/System.Data.Common/src/System/Data/DataTable.cs

```diff
@@ -442,7 +441,8 @@ internal void DeserializeTableSchema(SerializationInfo info, StreamingContext co
                 dc._columnUri = info.GetString(string.Format(formatProvider, "DataTable.DataColumn_{0}.Namespace", i));
                 dc.Prefix = info.GetString(string.Format(formatProvider, "DataTable.DataColumn_{0}.Prefix", i));
 
-                dc.DataType = (Type)info.GetValue(string.Format(formatProvider, "DataTable.DataColumn_{0}.DataType", i), typeof(Type));
+                string typeName = (string)info.GetValue(string.Format(formatProvider, "DataTable.DataColumn_{0}.DataType", i), typeof(string));
+                dc.DataType = Type.GetType(typeName);
```

Comment:
We should throw here if the type isn't resolvable: `Type.GetType(typeName, true);`

Reply:
@danmosemsft @stephentoub any strong opinion whether we should throw a TypeLoadException or a SerializationException here?

Reply:
I think the common pattern is a SerializationException with a nested exception (TypeLoadException in this case)